src/aspire/basis/ffb_2d.py

```diff
@@ -98,12 +98,12 @@ def evaluate(self, v):
         :param v: A coefficient vector (or an array of coefficient vectors)
             in FB basis to be evaluated. The first dimension must equal `self.count`.
         :return x: The evaluation of the coefficient vector(s) `x` in standard 2D
-            coordinate basis. This is an array whose first two dimensions equal `self.sz`
-            and the remaining dimensions correspond to dimensions two and higher of `v`.
+            coordinate basis. This is Image instance with resolution of `self.sz`
+            and the first dimension correspond to remaining dimension of `v`.
         """
-        # make should the first dimension of v is self.count
-        v, sz_roll = unroll_dim(v, 2)
```

Comment:
If it is the lines I was thinking, we should be okay because ` v = v.reshape(-1, self.count)` would leave the higher dims alone... effectively saying `(m,n, count)` is equiv to `(m*n, count)` for the purposes of subsequent nufft calls. I think this flattening would extend to any dim okay `(k,m,n, count) ~ (k*m*n, count)`.  Is that the behavior you were worried about?

Reply:
Yeah that's the behavior I'm thinking about. We (inadvertently) flatten any higher dimensions into one. The unrolling/rolling functions were there to preserve higher dimensions since some of the internal calls (like the nufft) aren't able to handle more than one extra dimension. By getting rid of those, we no longer preserve them. I think it's fine for now, since `Image` only handles one extra dimension, but we may want to bring that behavior back at a later point.

Reply:
It shouldn't be too hard, I did it once today for a different method. I'll try to do it here, and report back.

Reply:
Not bad. Pushed.

Reply:
Nice! At some later point, it might make sense to go through the function calls to see which ones don't play well with higher dimensions. My suspicion is that it's just `anufft`, provided we replace `:` with `...` in places. If we can push this rolling/unrolling stuff to `anufft`/`nufft`, it might simplify a lot of the basis code.

Reply:
Yes, makes sense to me, opened #260 for this.

Reply:
:pray: